python/ray/node.py

```diff
@@ -73,31 +82,47 @@ def __init__(self,
                 "workers/default_worker.py"))
 
         self._ray_params = ray_params
-
-        self._node_ip_address = ray_params.node_ip_address
         self._redis_address = ray_params.redis_address
         self._config = (json.loads(ray_params._internal_config)
                         if ray_params._internal_config else None)
 
+        self._init_temp()
+
+        if connect_only:
+            # If user does not provide the socket name, get it from Redis.
+            if (self._ray_params.plasma_store_socket_name is None
+                    or self._ray_params.raylet_socket_name is None):
+                # Get the address info of the processes to connect to
+                # from Redis.
+                address_info = ray.services.get_address_info_from_redis(
+                    self.redis_address,
+                    self._node_ip_address,
+                    redis_password=self.redis_password)
+                self._plasma_store_socket_name = address_info[
+                    "object_store_address"]
+                self._raylet_socket_name = address_info[
+                    "raylet_socket_name"]
+        else:
+            # If the user specified a socket name, use it.
+            self._plasma_store_socket_name = self._prepare_socket_file(
```

Comment:
I can't see why we don't remove `self._plasma_store_socket_name` and use `self._ray_params.plasma_store_socket_name` instead directly?

Reply:
This is because the role of `self._ray_pramas.raylet_socket_name` is the input configuration (not only for the node, and I think it is not a good idea to change it), but `self._plasma_store_socket_name` holds the actual state of the node (it could come from the ray params, the redis global state, 
 or it could be just generated by us). I have changed the code to fix the issue you mentioned before.